x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/searchablesnapshots/SearchableSnapshotShardStats.java

```diff
@@ -188,10 +191,39 @@ public CacheIndexInputStats(String fileName, long fileLength, long openCount, lo
             this.currentIndexCacheFills = in.readVLong();
         }
 
+        public static CacheIndexInputStats combine(CacheIndexInputStats cis1, CacheIndexInputStats cis2) {
+            if (cis1.getFileExt().equals(cis2.getFileExt()) == false) {
+                assert false : "can only combine same file extensions";
+                throw new IllegalArgumentException("can only combine same file extensions but was " +
+                    cis1.fileExt + " and " + cis2.fileExt);
+            }
+            return new CacheIndexInputStats(
+                cis1.fileExt,
+                cis1.numFiles + cis2.numFiles,
+                cis1.totalSize + cis2.totalSize,
+                cis1.openCount + cis2.openCount,
+                cis1.closeCount + cis2.closeCount,
+                cis1.forwardSmallSeeks.add(cis2.forwardSmallSeeks),
+                cis1.backwardSmallSeeks.add(cis2.backwardSmallSeeks),
+                cis1.forwardLargeSeeks.add(cis2.forwardLargeSeeks),
+                cis1.backwardLargeSeeks.add(cis2.backwardLargeSeeks),
+                cis1.contiguousReads.add(cis2.contiguousReads),
+                cis1.nonContiguousReads.add(cis2.nonContiguousReads),
+                cis1.cachedBytesRead.add(cis2.cachedBytesRead),
+                cis1.indexCacheBytesRead.add(cis2.indexCacheBytesRead),
+                cis1.cachedBytesWritten.add(cis2.cachedBytesWritten),
+                cis1.directBytesRead.add(cis2.directBytesRead),
+                cis1.optimizedBytesRead.add(cis2.optimizedBytesRead),
+                cis1.blobStoreBytesRequested.add(cis2.blobStoreBytesRequested),
+                cis1.currentIndexCacheFills + cis2.currentIndexCacheFills
+            );
+        }
+
         @Override
         public void writeTo(StreamOutput out) throws IOException {
-            out.writeString(fileName);
-            out.writeVLong(fileLength);
+            out.writeString(fileExt);
```

Comment:
These stats also exist in 7.11 with the old wire format so I think it'd be better to have a version guard here, rather than just assuming we're talking to a 7.12 node. Bailing out with an `UnsupportedOperationException` is ok, there's no need to support this API on a mixed cluster since it's only supposed to be for internal use.